app/controllers/topics_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TopicsController < ApplicationController
+  def index
+    @topics = Topic.all
+  end
+
+  def show
+    @topic = Topic.find(params[:id])
+  end
+
+  def new
+    @user = current_user
+    @topic = Topic.new
+  end
+  
+  def create
+    @user = current_user
+    @topic = @user.topics.build(topic_params)
+    
+    if @topic.save
+      redirect_to(topics_path)
+    else
+      flash[:alert] = 'There was an error creating the topic'
+      redirect_to(new_topic)
+    end
+  end
+
+  def edit
+    @topic = Topic.find(params[:id])
+  end
+  
+  def destroy
+    @user = current_user
+    @topic = @user.topics.find(params[:id])
```

Comment:
@ianroberts131 PS. Edited comment https://github.com/ianroberts131/blocmarks/pull/2/files#r64024819 to mention `edit` action only rather than all actions.